src/blocks/custom.rs

```diff
@@ -89,13 +109,26 @@ impl Block for Custom {
             .or_else(|| self.command.clone())
             .unwrap_or_else(|| "".to_owned());
 
-        let output = Command::new(env::var("SHELL").unwrap_or("sh".to_owned()))
+        let (statuscode, output) = Command::new(env::var("SHELL").unwrap_or("sh".to_owned()))
             .args(&["-c", &command_str])
             .output()
-            .map(|o| String::from_utf8_lossy(&o.stdout).trim().to_owned())
-            .unwrap_or_else(|e| e.description().to_owned());
+            .map(|o| (o.status.code().unwrap_or(254) , String::from_utf8_lossy(&o.stdout).trim().to_owned()))
+            .unwrap_or_else(|e| (255, e.description().to_owned()));
 
         self.output.set_text(output);
+        self.output.set_state( 
+                if self.critical_exit_codes.contains(&statuscode) {
+                    State::Critical
+                } else if self.warning_exit_codes.contains(&statuscode) {
+                    State::Warning
+                } else if self.good_exit_codes.contains(&statuscode) {
+                    State::Good
+                } else if self.info_exit_codes.contains(&statuscode) {
+                    State::Info
+                } else {
+                    State::Idle
+                }
+        );
```

Comment:
To improve performance in extreme cases (for example if the user configured a list of 1 million numbers), this could be changed to have a `HashMap<i32, State>` that could be populated in `Custom::new`.

Reply:
That seems like a silly usecase for us to support, so the existing approach makes more sense to me.